src/main/java/com/patchworkmc/mapping/remapper/NaiveRemapper.java

```diff
@@ -38,8 +53,14 @@ public void acceptMethodVar(IMappingProvider.Member method, int lvIndex, int sta
 
 			@Override
 			public void acceptField(IMappingProvider.Member field, String dstName) {
-				if (classes.get(field.name) != null) throw new IllegalArgumentException("Duplicated field name " + field.name);
-				fields.put(field.name, dstName);
+				String presentName = fields.get(field.name);
+
+				if (presentName == null || presentName.equals(dstName)) {
+					fields.put(field.name, dstName);
+					fieldDesc.put(dstName, field.desc);
+				} else {
+					Patchwork.LOGGER.debug("Duplicated field name %s (proposed %s, but key already mapped to %s!)", field.name, dstName, presentName);
```

Comment:
Why was this downgraded to a log message? Shouldn't we fail fast here?

Reply:
See above